debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresTaskContext.java

```diff
@@ -104,7 +104,9 @@ protected ReplicationConnection createReplicationConnection(boolean exportSnapsh
                             "will be created after a connector restart, resulting in missed data change events.",
                     PostgresConnectorConfig.DROP_SLOT_ON_STOP.name());
         }
-        return ReplicationConnection.builder(config.jdbcConfig())
+
+        return ReplicationConnection.builder(
+                config.jdbcConfig().edit().with("ApplicationName", "DBZ#" + config.getConfig().getString("name")).build())
```

Comment:
This looks ugly, but we can think about a different way to achieve it, but imo this is the right place to add that value.
Basically we set the application_name for that JDBC session to a specific value that includes the connector name. That's how we can identify if that was a new connector or an update case.

Reply:
I don't think it's that ugly actually; in fact we probably hsould set the application name like that for all those databases/drivers which support that. Ideally, the KC cluster name would also be part of this, but I reckon we cannot get hold of it? I'd only prefix it like "Debezium-...", for the sake of comprehensibility.

Reply:
As the cluster name is set via System Properties maybe we can get the value. I can try. Need to think about that multi-cluster example. Maybe I had this in mind when writing that validation logic, but I am not sure the logic works as expected.